cogs/Announce.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python
+
+"""
+Koala Bot Announce feature
+Created by: Bill Cao
+"""
+
+# Built-in/Generic Imports
+import asyncio
+
+# Libs
+
+import discord
+from discord.ext import commands
+from utils.KoalaUtils import extract_id
+from utils import KoalaColours
+import time
+
+# constants
+ANNOUNCE_SEPARATION_DAYS = 30
+
+import KoalaBot
+
+
+def announce_is_enabled(ctx):
+    """
+    A command used to check if the guild has enabled announce
+    e.g. @commands.check(announce_is_enabled)
+
+    :param ctx: The context of the message
+    :return: True if enabled or test, False otherwise
+    """
+    try:
+        result = KoalaBot.check_guild_has_ext(ctx, "Announce")
+    except PermissionError:
+        result = False
+
+    return result or (str(ctx.guild) == KoalaBot.TEST_USER and KoalaBot.is_dpytest)
+
+
+class Announce(commands.Cog):
+    """
+        A discord.py cog to allow announcements to certain roles.
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.messages = {}
+        self.roles = {}
+        KoalaBot.database_manager.create_base_tables()
+        KoalaBot.database_manager.insert_extension("Announce", 0, True, True)
+        self.announce_database_manager = AnnounceDBManager(KoalaBot.database_manager)
+        self.announce_database_manager.create_tables()
+
+    def not_exceeded_limit(self, guild_id):
+        """
+        Check if enough days have passed for the user to use the announce function
+        :return:
+        """
+        if self.announce_database_manager.get_last_use_date(guild_id):
+            return int(time.time()) - self.announce_database_manager.get_last_use_date(
+                guild_id) > ANNOUNCE_SEPARATION_DAYS * 24 * 60 * 60
```

Comment:
Consider changing this / adding a constant called `ANNOUNCE_SEPARATION_SECONDS` to remove magic numbers in code